### PR TITLE
[sui-execution] Introduce Verifier interface

### PR DESCRIPTION
**Pull request by amnn**
## Description

This interface is used to perform metered verification during signing (verification during execution is handled internally, within the execution layer).

To fully encapsulate metering within the execution layer, some tests needed to be moved from `sui-core`  to `sui-framework-tests` (which is linked against the latest execution layer, so therefore can access internal types).

This change also gets rid of the distinction between metered verification of module bytes and of deserialized `CompiledModules` within the execution layer, to reduce the interface surface area.  Instead, a helper function has been added to the interface to first deserialize the module.

## Test Plan

```
cargo simtest
env SUI_SKIP_SIMTESTS=1 cargo nextest run
```

### CODE REVIEW

File: crates/sui-framework-tests/src/lib.rs [hunk 1]
```diff
@@ -70,6 +77,345 @@ mod test {
         });
     }
 
+    #[test]
+    #[cfg_attr(msim, ignore)]
+    fn test_metered_move_bytecode_verifier() {
```
Comment (amnn) [inline crates/sui-framework-tests/src/lib.rs:82]: Tests moved from `move_package_tests` to avoid exposing `Scope` and `Meter` types via `sui-execution`.

Comment (dariorussi) [reply crates/sui-framework-tests/src/lib.rs:82]: why not in its own file? I guess because `lib.rs` is the only file in this directory?

**Review by dariorussi — APPROVED** [review]
it looks pretty clean

File: crates/sui-framework-tests/src/lib.rs (hunk 1 above)
Comment (amnn) [reply crates/sui-framework-tests/src/lib.rs:82]: Yeah, it's a fair question. I was just doing this mechanically, but a new file sounds good.

**Pull request merged**